src/lib/http/HttpError.ts

```diff
@@ -0,0 +1,13 @@
+import { HttpRequest } from './HttpRequest';
+import { HttpResponse } from './HttpResponse';
+
+export class HttpError extends Error {
+  constructor(
+    message: string,
+    public readonly request?: HttpRequest,
+    public readonly response?: HttpResponse
+  ) {
+    super(message);
+    Object.setPrototypeOf(this, new.target.prototype);
```

Comment:
required if we want to support 'instanceof'  checks as we are compiling down to es5